src/whitenoise/groups.rs

```diff
@@ -464,6 +482,72 @@ impl Whitenoise {
         Ok(())
     }
 
+    /// Uploads a group profile picture to a Blossom server.
+    ///
+    /// This method generates new keys for the upload, uploads the image file to the specified
+    /// Blossom server, and returns the hash, encryption key, and nonce that can be used to
+    /// update the group configuration data. The private key from the generated keys should
+    /// be stored in the NostrGroupConfigData.
+    ///
+    /// # Arguments
+    /// * `file_path` - Path to the image file to upload
+    /// * `image_type` - Image type (JPEG, PNG, etc.)
+    /// * `server` - Blossom server URL
+    pub async fn upload_group_profile_picture(
+        &self,
+        file_path: &str,
+        image_type: ImageType,
+        server: Url,
+    ) -> Result<GroupProfilePictureUpload> {
+        let client = BlossomClient::new(server);
+        let data = tokio::fs::read(file_path).await?;
+
+        // Generate new keys for this upload
+        let keys = Keys::generate();
+
+        // Upload the file
+        let descriptor = client
+            .upload_blob(
+                data,
+                Some(image_type.mime_type().to_string()),
+                None,
+                Some(&keys),
+            )
+            .await
+            .map_err(|err| WhitenoiseError::Other(anyhow::anyhow!(err)))?;
+
+        // Use the SHA256 hash from the BlobDescriptor
+        let hash_hex = &descriptor.sha256;
+        let hash_bytes = hex::decode(hash_hex)
+            .map_err(|e| WhitenoiseError::Other(anyhow::anyhow!("Failed to decode hash: {}", e)))?;
+
+        // Ensure the decoded hash is exactly 32 bytes (SHA256 length)
+        if hash_bytes.len() != 32 {
+            return Err(WhitenoiseError::Other(anyhow::anyhow!(
+                "Invalid hash length: expected 32 bytes, got {} bytes",
+                hash_bytes.len()
+            )));
+        }
+
+        let mut image_hash = [0u8; 32];
+        image_hash.copy_from_slice(&hash_bytes);
```

Comment:
Should we compute the sha256 ourselves and compare that Blossom gives us the proper value? (this is something that could be done directly on the BlossomClient but as far as I can tell it's currently not done)